internal/hardware/validator.go

```diff
@@ -318,21 +319,40 @@ func totalizeRequirements(ocpRequirements models.ClusterHostRequirementsDetails,
 	return total
 }
 
-func (v *validator) getOCPClusterHostRoleRequirementsForVersion(cluster *common.Cluster, role models.HostRole) (models.ClusterHostRequirementsDetails, error) {
+func (v *validator) getOCPClusterHostRoleRequirementsForVersion(cluster *common.Cluster, host *models.Host) (models.ClusterHostRequirementsDetails, error) {
 	requirements, err := v.getOCPRequirementsForVersion(cluster.OpenshiftVersion)
 	if err != nil {
 		return models.ClusterHostRequirementsDetails{}, err
 	}
 
-	if role == models.HostRoleMaster {
+	if common.GetEffectiveRole(host) == models.HostRoleMaster {
 		if common.IsSingleNodeCluster(cluster) {
 			return *requirements.SNORequirements, nil
 		}
 		return *requirements.MasterRequirements, nil
 	}
+
+	if v.isEdgeWorker(host) {
+		return *requirements.EdgeWorkerRequirements, nil
+	}
+
 	return *requirements.WorkerRequirements, nil
 }
 
+// There is no need to fail here, failed to get inventory just return false
+func (v *validator) isEdgeWorker(host *models.Host) bool {
+	inventory, err := common.UnmarshalInventory(host.Inventory)
+	if err != nil {
+		return false
+	}
+	if inventory.CPU.Architecture != common.ARM64CPUArchitecture {
+		return false
+	}
+
+	edgeWorkersProductList := strings.Split(strings.ToLower(strings.ReplaceAll(v.EdgeWorkerProductNames, " ", "")), ",")
```

Comment:
consider doing this part once when you initiate the validator, otherwise you will run it in the monitor all the time for each host.